libafl/src/fuzzer/mod.rs

```diff
@@ -598,9 +632,14 @@ where
         #[cfg(feature = "introspection")]
         state.introspection_monitor_mut().reset_stage_index();
 
+        let pre_corpus_size = state.corpus().count();
         // Execute all stages
         stages.perform_all(self, executor, state, manager, idx)?;
 
+        let end_corpus_size = state.corpus().count();
+
+        *state.own_finds_mut() = end_corpus_size - pre_corpus_size;
```

Comment:
`own_finds` is state.corpus() - state.imported(), no?

Reply:
so the basic idea is,

this file fuzzer.rs has the code that every fuzzer will runs no matter what the fuzzer's composition is.
we don't to add code unless it is really needed by every fuzzer.

for this own_finds_mut, this is something that you can already compute in maybe_report_progress,
so better to move this code to that place.

Reply:
> `own_finds` is state.corpus() - state.imported(), no?

Nope, `own_finds` is the number of new paths found during this fuzzing section according to the documentation of AFL. Note that during this fuzzing section. So `own_finds` is not a cumulative value.

Reply:
> so the basic idea is,
> 
> this file fuzzer.rs, has the code that every fuzzer will runs no matter what the fuzzer's composition is. we don't want to add code here in this function unless it is really needed by every fuzzer.
> 
> for this own_finds_mut, this is something that you can already compute in maybe_report_progress, so better to move this code to that place.

How about handling `own_finds` in the function `MutationalStage.perform_mutational()`?

Reply:
> Nope, own_finds is the number of new paths found during this fuzzing section according to the documentation of AFL.

but for example if you run afl++ on some target you'll see own_finds is close to the number of corpus count. so it is a cumulative value.

so I guess this is `state.corpus() - <the number of initial corpus> - state.imported()`
this number is `afl->queued_discovered` in afl++. you can look at how this number is counted

Reply:
Well, I have gone though the source code of AFL. You are right!